### DISCUSSION TITLE
ZOS 3.0 BUG - MiniApp auto opens when sleeping

### DISCUSSION THREAD

**Discussion by AFumi39** [zepp-health/zeppos-forum#277] [User Feedback]
Hello, I have this annoying bug that's quite strange.
I have an Amazfit Balance with Zepp OS 3.0 and SW Version 3.15.10.1
I installed via Developer Mode a simple Mini Program (see the attachments) that have a button and do an httpRequest when pressed.
The Mini Program works well but...

That's very strange, only when I'm sleeping (does not happen during the rest of the day or after the "morning updates"), the Mini Program is automatically launched (many instances of the program, that I have to close one by one). This behavior causes a huge drop in battery life: the display turns on every time, like it's a notification, and turns on every time I move the wrist (until I close all the instances).

I attach a video of some instances that I had to close last night. I didn't turned the screen on, it automatically turned on by the "auto instance open".

If I uninstall the Mini Program, the problem doesn't occurs.

Anyone have a suggestion to fix this?

Thank you

https://github.com/zepp-health/zeppos-forum/assets/85888564/6d8aa7af-155a-49ce-9c3e-b68a5e8e6e46

[hello-world-test3.zip](https://github.com/zepp-health/zeppos-forum/files/13598778/hello-world-test3.zip)

**Comment by MyoungXUE** [comment]
Hello, I have looked at your code and it seems that there should not be too many problems. The code will not cause the problem you mentioned, and the problem needs to be located from the log.

Please submit user feedback in Zepp and check the upload log option. Relevant colleagues will analyze this problem. At the same time, please tell me your user ID, which will help us locate the problem faster.

**Comment by AFumi39** [comment]
Thank you, my user ID is 3008515569 (just sent the feedback, yesterday I sent another one regarding PAI, that is not urgent).

I add that I had the problem for few nights, so yesterday morning I tried to factory reset the Balance, and install the Mini Program again. That was useless because the problem occurred again last night.

I attach another video, that I filmed the night before. at the end you can see that the Program starts itself without any interaction by myself.

Many thanks!

https://github.com/zepp-health/zeppos-forum/assets/85888564/119a4899-9b43-44aa-aa25-d83efdb20a36

**Reply by MyoungXUE** [reply]
Hello, I have seen the relevant logs. Thank you for your cooperation. The relevant colleagues are investigating and I will notify you if there is progress.

**Comment by zhouxinhe** [comment]
@AFumi39
Thanks for your feedback!
I'm sorry to tell you that it is a software bug, and we will fix it ASAP.
Before that, you can remove the "event:os.health.sleep_status" permission from your program's app.json.

**Reply by AFumi39** [reply]
Thank you, I'll try to remove the event 👍
